### ISSUE TITLE
500.31 ANCM Failed to Find Native Dependencies - additional requirement

### ISSUE THREAD

**Issue by OnenessUK** [dotnet/AspNetCore.Docs#21937]
It might be worth adding to this section the fact that 500.31 also occurs on IIS if the **Startup object** isn't specified in the project file.

I wasted a lot of time not knowing that one one.

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4a2a7104-1d14-2e8a-0984-d92aa311c108
* Version Independent ID: 5e3c0546-c579-afb9-5289-b14532c127eb
* Content: [Troubleshoot ASP.NET Core on Azure App Service and IIS](https://docs.microsoft.com/en-gb/aspnet/core/test/troubleshoot-azure-iis?view=aspnetcore-5.0)
* Content Source: [aspnetcore/test/troubleshoot-azure-iis.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/test/troubleshoot-azure-iis.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-test**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @OnenessUK ... Can you clarify what you mean by ...

> the Startup object isn't specified in the project file

**Comment by guardrex** [comment]
@OnenessUK ... We're going to wait and take more feedback before adding a troubleshooting item for that particular problem. I understand it's one of those *pain-to-figure-out-from-the-error* 😠 types of errors, but there are hundreds of rare error scenarios like that. If the product unit or the docs cats receive more feedback on this, then we can add something.

I recommend that you flesh out what you mean about the error condition. A dev looking at all of the issues attached to this topic (or via search) will probably find this issue, so it will document the error scenario for them even if we don't put it into the topic text.

Can you show what you mean exactly? ... I assume that you're saying that your web host builder in `ConfigureWebHostDefaults` didn't set a `Startup` for `UseStartup` or some such. Idk how the app would run locally tho if that's what your app has. Anyway ... whatever the scenario ... I recommend that you paste the code in here and clarify if this is local, Azure, IIS, etc. that produces the error. Perhaps, show the fix, too, if it isn't just what I said about setting the `Startup` class :point_up:.

**Comment by OnenessUK** [comment]
Running my .NET Core 5 app on IIS, I was getting error : 500.31.

I noticed the Startup object was empty, in the project properties in VS, Application tab. When I changed (Not set) in the drop down to my application Program file, DataApi.Program, it worked.

I believe selecting a value adds it to the project file:

<PropertyGroup>
    ...
    <StartupObject>DataApi.Program</StartupObject>
  </PropertyGroup>

This was all on my local development machine (I use IIS as well as IISExpress).

**So changing <StartupObject> on my local machine definitely solved the 500.31 error issue, hence my feedback.**

However, on our remote staging server, I'm still getting 500.31, even with the <StartupObject> correctly set. So maybe there something else going on here.

Having a frustrating time trying to get it working, the hosting bundle and SDK are correctly installed on the server (Windows Server 2016 Datacenter) but getting this 500.31 , the event log states: "failed to load coreclr". I've allowed 32  bit apps in the application pool, suspecting it's a x64/x86 issue, but don't know where to go from here.

It runs fine using IIS Express on this staging server.

**Comment by guardrex** [comment]
@OnenessUK ... Ah ... I see. That is a highly unusual scenario/error. I've been around here since the beta 5 era, and this particular error condition has never come up IIRC. Open this for the product unit. It's very likely that they can help. Give them a good repro setup if you can, as they might need to try and reproduce it there.

https://github.com/dotnet/aspnetcore/issues

Please add a `cc: @guardrex` to your opening comment so that I can follow the discussion. I'll re-open this if it looks like there's something to cover. Also, cross-linking this issue to the PU issue will help future readers find your PU issue even if we don't add something to the doc. Here's a cut-'n-paste that you can add to the bottom of your opening comment over there ...

```
Cross-reference: https://github.com/dotnet/AspNetCore.Docs/issues/21937
cc: @guardrex
```
